src/workarounds/forwarddiff_rules.jl

```diff
@@ -232,12 +233,22 @@ function self_consistent_field(basis_dual::PlaneWaveBasis{T};
     eigenvalues = map(scfres.eigenvalues, getfield.(δresults, :δeigenvalues)...) do εk, δεk...
         map((εnk, δεnk...) -> DT(εnk, δεnk), εk, δεk...)
     end
+    occupation = map(scfres.occupation, getfield.(δresults, :δoccupation)...) do occk, δocck...
+        map((occnk, δoccnk...) -> DT(occnk, δoccnk), occk, δocck...)
+    end
+    εF = DT(scfres.εF, getfield.(δresults, :δεF)...)
 
     # TODO Could add δresults[α].δVind the dual part of the total local potential in ham_dual
     # and in this way return a ham that represents also the total change in Hamiltonian
 
-    merge(scfres, (; ψ, ρ, eigenvalues, basis=basis_dual,
-                   response=getfield.(δresults, :history)))
+    # TODO use dual occupation below (or not?)
+    energies, ham = energy_hamiltonian(basis_dual, ψ, scfres.occupation; ρ, eigenvalues, εF)
+
+    # TODO return dual occupation
+    merge(scfres, (; ψ, ρ, eigenvalues, energies, εF,
```

Comment:
Avoid the merge as Antoine mentioned.

Reply:
done [c17293b](https://github.com/JuliaMolSim/DFTK.jl/pull/709/commits/c17293bb8035210caf358a1d35419a6093563bdf)